lib/netext/dialer.go

```diff
@@ -56,11 +56,15 @@ func (d Dialer) DialContext(ctx context.Context, proto, addr string) (net.Conn,
 	}
 	if v := ctx.Value(ctxKeyTracer); v != nil {
 		tracer := v.(*Tracer)
-		return &Conn{conn, &tracer.bytesRead, &tracer.bytesWritten}, nil
+		return d.TraceConnection(conn, &tracer.bytesRead, &tracer.bytesWritten)
 	}
 	return conn, err
 }
 
+func (d Dialer) TraceConnection(conn net.Conn, bytesRead, bytesWritten *int64) (*Conn, error) {
```

Comment:
I feel like having this under a Dialer + the name makes it rather confusing what this function actually does. It should be a free function, maybe called `TrackConn()` or something like that? Probably placed right under Conn, like a constructor.